packages/jest-cli/src/SnapshotInteractiveMode.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @flow
+ */
+
+import type {AggregatedResult} from 'types/TestResult';
+
+const chalk = require('chalk');
+const ansiEscapes = require('ansi-escapes');
+const {pluralize} = require('./reporters/utils');
+const {rightPad} = require('./lib/terminalUtils');
+const {KEYS} = require('./constants');
+
+module.exports = class SnapshotInteractiveMode {
+  _pipe: stream$Writable | tty$WriteStream;
+  _isActive: boolean;
+  _updateTestRunnerConfig: (a: string, jestRunnerOptions: Object) => *;
+  _testFilePaths: Array<string>;
+  _countPaths: number;
+
+  constructor(pipe: stream$Writable | tty$WriteStream) {
+    this._pipe = pipe;
+    this._isActive = false;
+  }
+
+  isActive() {
+    return this._isActive;
+  }
+
+  _drawUIOverlay() {
+    this._pipe.write(ansiEscapes.scrollDown);
+    this._pipe.write(ansiEscapes.scrollDown);
+
+    this._pipe.write(ansiEscapes.cursorSavePosition);
+    this._pipe.write(ansiEscapes.cursorTo(0, 0));
+
+    const title = rightPad(' -> Interactive Snapshot Update Activated <-');
+    this._pipe.write(chalk.black.bold.bgYellow(title));
+
+    this._pipe.write(ansiEscapes.cursorRestorePosition);
+    this._pipe.write(ansiEscapes.cursorUp(6));
+    this._pipe.write(ansiEscapes.eraseDown);
+
+    const numFailed = this._testFilePaths.length;
+    const numPass = this._countPaths - this._testFilePaths.length;
+
+    let stats = chalk.bold.red(pluralize('suite', numFailed) + ' failed');
+    if (numPass) {
+      stats += ', ' + chalk.bold.green(pluralize('suite', numPass) + ' passed');
+    }
+    const messages = [
+      '\n' + chalk.bold('Interactive Snapshot Progress'),
+      ' \u203A ' + stats,
+      '\n' + chalk.bold('Watch Usage'),
+      chalk.dim(' \u203A Press ') +
+        'u' +
+        chalk.dim(' to update failing snapshots.'),
```

Comment:
I'd rather like something more descriptive:
`Press u to update failing snapshots for this test`.